File: src/main/java/com/dyonovan/neotech/common/blocks/connected/BlockPhantomGlass.java
package com.dyonovan.neotech.common.blocks.connected;

import com.dyonovan.neotech.helpers.GuiHelper;
import net.minecraft.block.material.Material;
import net.minecraft.block.state.IBlockState;
import net.minecraft.entity.Entity;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.AxisAlignedBB;
import net.minecraft.util.BlockPos;
import net.minecraft.world.World;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import org.lwjgl.input.Keyboard;

import java.util.List;

public class BlockPhantomGlass extends BlockConnectedTextures {
    public BlockPhantomGlass(Material materialIn, String name, Class<? extends TileEntity> tileClass) {
        super(materialIn, name, tileClass);
        setHardness(1.5F);
    }

    @Override
    public void addCollisionBoxesToList(World worldIn, BlockPos pos, IBlockState state, AxisAlignedBB mask, List list, Entity collidingEntity) {
        if (((collidingEntity instanceof EntityPlayer)) && (!collidingEntity.isSneaking())) {
            return;
        }
        super.addCollisionBoxesToList(worldIn, pos, state, mask, list, collidingEntity);
    }

    @SideOnly(Side.CLIENT)
    public boolean isTranslucent() {
        return true;
    }

    @Override
    public boolean isOpaqueCube() {
        return false;
    }

    @Override
    public void addToolTip(List<String> toolTip) {
        if(Keyboard.isKeyDown(Keyboard.KEY_LSHIFT) || Keyboard.isKeyDown(Keyboard.KEY_RSHIFT)) {
            toolTip.add(GuiHelper.GuiColor.YELLOW + "Is solid to all entities except players");
            toolTip.add("Sneaking removes this effect");
        } else {
            toolTip.add(GuiHelper.GuiColor.GRAY + "Shift for More Info");
        }
    }
}

File: src/main/scala/com/dyonovan/neotech/common/blocks/BlockPhantomGlass.scala
package com.dyonovan.neotech.common.blocks

import com.dyonovan.neotech.NeoTech
import com.dyonovan.neotech.lib.Reference
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraft.block.state.IBlockState
import net.minecraft.entity.Entity
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.{AxisAlignedBB, BlockPos}
import net.minecraft.world.World

/**
  * This file was created for NeoTech
  *
  * NeoTech is licensed under the
  * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
  * http://creativecommons.org/licenses/by-nc-sa/4.0/
  *
  * @author Paul Davis "pauljoda"
  * @since 2/26/2016
  */
class BlockPhantomGlass extends Block(Material.glass) with BlockConnectedTextures {

    setCreativeTab(NeoTech.tabDecorations)
    setUnlocalizedName(Reference.MOD_ID + ":phantomGlass")

    /**
      * Define true if you are a clear texture
      *
      * @return
      */
    override def isClear: Boolean = true

    // Methods to move textures to lower class, handle others here
    override def NoCornersTextureLocation: String = "neotech:blocks/connected/phantomGlass/phantomGlass"
    override def CornersTextureLocation: String = "neotech:blocks/connected/phantomGlass/phantomGlass_corners"
    override def VerticalTextureLocation: String = "neotech:blocks/connected/phantomGlass/phantomGlass_vertical"
    override def AntiCornersTextureLocation: String = "neotech:blocks/connected/phantomGlass/phantomGlass_anti_corners"
    override def HorizontalTextureLocation: String = "neotech:blocks/connected/phantomGlass/phantomGlass_horizontal"

    override def addCollisionBoxesToList(worldIn: World, pos: BlockPos, state: IBlockState, mask: AxisAlignedBB,
                                            list: java.util.List[AxisAlignedBB], collidingEntity: Entity) : Unit = {
        if(collidingEntity.isInstanceOf[EntityPlayer] && !collidingEntity.isSneaking) { /* NO OP */ }
        else
            super.addCollisionBoxesToList(worldIn, pos, state, mask, list, collidingEntity)
    }
}
